django/db/models/operators.py

```diff
@@ -0,0 +1,20 @@
+from django.db.models.expressions import CombinedExpression
+from django.db.models.fields import (
+    DateField, DateTimeField, DurationField, TimeField
+)
+
+__all__ = ()
+
+
+@CombinedExpression.dispatch(CombinedExpression.SUB, DateField, DateField)
+@CombinedExpression.dispatch(CombinedExpression.SUB, DateTimeField, DateTimeField)
+@CombinedExpression.dispatch(CombinedExpression.SUB, TimeField, TimeField)
```

Comment:
This could end up being problematic as only subtraction of same temporal types is actually supported.